### ISSUE TITLE
The new user templates for groups

### ISSUE THREAD

**Issue by jk164** [nextcloud/server#18326]
The group admins concept is very usefull, but limited in the current implementation. Existing limits are for example:
- Superadmin is unable to set Group Admin limits to quota settings (e.g. issue  #7429)
- Group admin is not limited is not limited when choosing a username for new user (limited protection of user namespace consistency)
- Superadmin cannot predefine new user membership in other groups (e.g. a general group for restriction of usual users to reshare data).

A possible solution to consider is to deploy a group template (limits and predefined values) for every new group. Such a template would contain informations that would limit the group admin when creating a new user or editing current user of his group. No admin (Group admin or Superadmin) could exceed the limits specified by the group template unless changed by the Superadmin. The template could include for  example:
- maximum quota that the admin is allowed to set for any group user (0 - the admin is not allowed to set any quota)
- fix username prefix/sufix (e.g. the group name as a fix prefix of the final username). The group admin define "User1" in the group with "Group1_" username prefix -> the inal username of user is "Group1_User1" .
- The Superadmin-defined membership in other groups that will be assigned to each new user.

The default group template will not contain any restrictions. The group template can have a design very close to the line for entering a new (editing existing) user.

thanks for consideration!

**Comment by bpcurse** [comment]
We haven't been able to use group admins for years because of the above mentioned issues. This proposal hits the bull's-eye and would finally render this feature usable for us.

**Comment by manu5801** [comment]
I completely agree with the proposal: our platform receives different associations, each association is independant of the others. There is one group for one association, and the subadmin functionality is very important for us as it gives autonomy to the associations. But the quota are discussed at beginning, and they should NOT be changed by the subadmins.
